lib/bibliothecary/analyser.rb

```diff
@@ -89,12 +89,11 @@ def parse_json_runtime_manifest(file_contents)
       end
 
       def map_dependencies(hash, key, type)
-        hash.fetch(key,[]).map do |name, requirement|
-          {
-            name: name,
-            requirement: requirement,
-            type: type
-          }
+        # determine if the items are hashes or arrays, and return appropriate results
```

Comment:
I think we can keep this internal to `Conda` by removing this change and just having `Conda` do the right thing

Reply:
I kind of disagree. Yes there's the YAGNI principle, but keeping this in will allow future parsers to not care whether they are a two-tuple of `(name, requirement)` or a nicely formatted response from a json-api.

(for instance, with this in, I can use it in `npm.rb`, `hackage.rb`, `hex.rb`, `maven.rb`, etc..  as well, rather than having  almost the same maps to maintain there.
I might do this in a new PR after this, I think it's out of scope of this PR, but will be a small change)